lib/src/native/realm_core.dart

```diff
@@ -47,9 +48,26 @@ import '../user.dart';
 import '../set.dart';
 import 'realm_bindings.dart';
 
-late RealmLibrary _realmLib;
+part 'decimal128.dart';
 
-final _RealmCore realmCore = _RealmCore();
+const bugInTheSdkMessage = "This is likely a bug in the Realm SDK - please file an issue at https://github.com/realm/realm-dart/issues";
+
+final _realmLib = () {
+  final result = RealmLibrary(initRealm());
+  nativeLibraryVersion = result.realm_dart_library_version().cast<Utf8>().toDartString();
+  if (libraryVersion != nativeLibraryVersion) {
+    final additionMessage =
+        isFlutterPlatform ? bugInTheSdkMessage : "Did you forget to run `dart run realm_dart install` after upgrading the realm_dart package?";
+    throw RealmException('Realm SDK package version does not match the native library version ($libraryVersion != $nativeLibraryVersion). $additionMessage');
+  }
+  return result;
+}();
+
+// stamped into the library by the build system (see prepare-release.yml)
+const libraryVersion = '1.0.3';
+late String nativeLibraryVersion;
```

Comment:
It seems we don't need this field. So 
>nativeLibraryVersion = result.realm_dart_library_version().cast<Utf8>().toDartString();
could be a local variable.

Reply:
Also If I do `dart test -n "devide" -r expanded` in the repo the `decimal` tests are not included in the run at all.

Reply:
Try `dart test -n 'divide' -r expanded` instead.